tests/unit/components/UrlBar.spec.js

```diff
@@ -21,6 +21,72 @@ describe("Given a UrlBar", () => {
       expect(urlBar.find("input").element.placeholder).to.equal(placeholder);
     });
   });
+
+  describe("When a valid URL is typed into the UrlBar", () => {
+    let input;
+
+    beforeEach(() => {
+      input = urlBar.find("input");
+
+      input.element.value = "http://www.test.com";
+    });
+
+    describe("And the <Enter> key is pressed in the UrlBar", () => {
+      beforeEach(() => {
+        input.trigger("keyup.enter");
+      });
+
+      it("Then the input element should be registered as valid", () => {
+        const isInputValid = !input.element.classList.contains("invalid");
+
+        expect(isInputValid).to.be.true;
+      });
+    });
+  });
+
+  describe("When an invalid URL is typed into the UrlBar", () => {
+    let input;
+
+    beforeEach(() => {
+      input = urlBar.find("input");
+
+      input.element.value = "abc";
+    });
+
+    describe("And the <Enter> key is pressed in the UrlBar", () => {
+      beforeEach(() => {
+        input.trigger("keyup.enter");
+      });
+
+      it("Then the input element should be registered as invalid", () => {
+        const isInputValid = !input.element.classList.contains("invalid");
+
+        expect(isInputValid).to.be.false;
+      });
+
+      it("Then no request is made", () => {
```

Comment:
This doesn't make sense. The UrlBar has nothing to do with making a request. Nor does your test have anything to do with requests.

Reply:
Changed name to `"Then no event is emitted"`